### ISSUE TITLE
Memory leak at looks_like_sereal

### ISSUE THREAD

**Issue by kirill-binary** [Sereal/Sereal#230]
Perl version: perl 5, version 26, subversion 2 (v5.26.2) built for x86_64-linux
Package version Sereal-Decoder-4.011

I noticed that a subroutine `looks_like_sereal` is leaking.

Probably problem in this  line https://metacpan.org/source/YVES/Sereal-Decoder-4.011/Decoder.xs#L155

Script to reproduce the problem:
```
use strict;
use Sereal qw(looks_like_sereal);
use Sereal::Encoder qw(encode_sereal sereal_encode_with_object);
my $encoder = Sereal::Encoder->new();
my $out = $encoder->encode({test => 1});
while (1) { looks_like_sereal($out); }  
```

**Comment by kirill-binary** [comment]
PR for this issue https://github.com/Sereal/Sereal/pull/231

**Comment by demerphq** [comment]
Ill get this applied, but since it is failing travis-ci right now im going to investigate further. I might just rebase the patches into a different order and see how travis-ci responds, but im on the wrong computer right now to take care of it.

**Comment by demerphq** [comment]
Fixed in 4.012, thanks for the report and patches!
